ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -110,6 +110,11 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
     onTaskResume()
   }
 
+  override fun onPause() {
+    super.onPause()
+    saveCurrentState()
+  }
```

Comment:
Would it cause the value to be saved twice when pressing "next" or "previous" button? Can we make it more efficient?

Reply:
hm..earlier I was planning to control from parent fragment to avoid this condition, as I was not thinking to handle using any extra param.
let me try some other approach.

Reply:
fixed

Reply:
If we are saving the state in `onPause`, then do we even need to save it when pressing "previous" or "next" button?

Reply:
as per the current logic, our function will not work in onPause and onNext and onPrevious has there own way of saying and emiting the data, if we bring all functions under one roof, lot of checks will require, so I choose the each one has there own way of doing things and responsibility